Source/levels/drlg_l1.cpp

```diff
@@ -954,10 +954,10 @@ void GenerateRoom(int x, int y, int w, int h, int dir)
 
 void FirstRoom()
 {
-	VerticalLayout = GenerateRnd(2) == 0;
-	HasChamber1 = (GenerateRnd(2) != 0);
-	HasChamber2 = (GenerateRnd(2) != 0);
-	HasChamber3 = (GenerateRnd(2) != 0);
+	VerticalLayout = !FlipCoin();
+	HasChamber1 = FlipCoin();
+	HasChamber2 = FlipCoin();
+	HasChamber3 = FlipCoin();

```

Comment:
All though it seams strange that VerticalLayout uses an inverted check, I think you will agree that it's better then the original code :)

https://github.com/diasurgical/devilution/blob/master/Source/drlg_l1.cpp#L1486

Reply:
There we go: https://github.com/diasurgical/devilutionX/pull/4912